### PR TITLE
Implement firebase login

### PR DESCRIPTION
**Pull request by goncy**

### CODE REVIEW

File: src/components/ui/Avatar/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+
+import User from 'components/ui/Icons/User';
+
+// Styled components
+import { Container } from './styled';
+
+const Avatar = ({ src, ...props }) => (src ? <Container src={src} {...props} /> : <User {...props} />);
```
Comment (lndgalante) [inline src/components/ui/Avatar/index.js:9]: Can we add `src` PropTypes?

File: src/components/ui/Avatar/styled.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import styled from 'styled-components';
+
+export const Container = styled.img`
+  width: 36px;
+  height: 36px;
+  border: 2px solid ${({ theme }) => theme.text};
+  border-radius: 50%;
+`;
```
Comment (lndgalante) [inline src/components/ui/Avatar/styled.js]: Can we use `rem` units through polished rem util?

Example:
```javascript
import { rem } from 'polished';

export const Container = styled.img`
  width: ${rem('36px')};
  height: ${rem('36px')};
  border: 2px solid ${({ theme }) => theme.text};
  border-radius: 50%;
`;
```

File: src/components/ui/Icons/User.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+const User = ({ theme, ...props }) => (
+  <svg height="36px" viewBox="0 0 1000 1000" width="36px" {...props}>
+    <g>
+      <path
+        d="M500,10C228.8,10,10,228.8,10,500c0,271.3,218.8,490,490,490c271.3,0,490-218.8,490-490C990,228.8,771.3,10,500,10z M804.5,822C794,657.5,619,613.8,619,613.8s106.8-71.8,68.3-217c-19.3-75.3-91-131.3-189-131.3c-98,0-169.7,56-189,131.3c-38.5,145.2,68.2,217,68.2,217S202.5,652.3,192,822C109.7,739.7,57.3,626,57.3,500C57.3,255,255,57.3,500,57.3C745,57.3,942.8,255,942.8,500C942.8,626,890.2,739.7,804.5,822z"
+        fill={theme.text}
+      />
+    </g>
+  </svg>
+);
```
Comment (lndgalante) [inline src/components/ui/Icons/User.js]: I've just minified this SVG through [svgr](https://react-svgr.com/playground/) tool

```javascript
const User = ({ theme, ...props }) => (
    <svg height={36} viewBox="0 0 1000 1000" width={36} {...props}>
      <path fill={theme.text} d="M500 10C228.8 10 10 228.8 10 500c0 271.3 218.8 490 490 490 271.3 0 490-218.8 490-490S771.3 10 500 10zm304.5 812C794 657.5 619 613.8 619 613.8s106.8-71.8 68.3-217c-19.3-75.3-91-131.3-189-131.3s-169.7 56-189 131.3c-38.5 145.2 68.2 217 68.2 217S202.5 652.3 192 822C109.7 739.7 57.3 626 57.3 500 57.3 255 255 57.3 500 57.3S942.8 255 942.8 500c0 126-52.6 239.7-138.3 322z" />
    </svg>
);
```

File: src/session/context.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  React.useEffect(
+    () =>
+      auth.onChange((user) => {
+        setUser(user);
+
+        setStatus('restored');
+      }),
+    [],
+  );
```
Comment (lndgalante) [inline src/session/context.js]: Can we have an object for possible `status` values?

Example:
```javascript
const STATUS = {
  init: 'init',
  restored: 'restored'
}
```

File: src/utils/firebase/client.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import firebase from 'firebase/app';
+
+import 'firebase/auth';
+import 'firebase/firestore';
+
+export default new Proxy(
```
Comment (lndgalante) [inline src/utils/firebase/client.js:16]: Whoa why is this a Proxy?

File: src/utils/firebase/client.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+      if (!firebase.apps.length) {
+        firebase.initializeApp({
+          apiKey: process.env.GATSBY_FIREBASE_API_KEY,
+          authDomain: process.env.GATSBY_FIREBASE_AUTH_DOMAIN,
+          databaseURL: process.env.GATSBY_FIREBASE_DATABASE_URL,
+          projectId: process.env.GATSBY_FIREBASE_PROJECT_ID,
+          storageBucket: process.env.GATSBY_FIREBASE_STORAGE_BUCKET,
+          messagingSenderId: process.env.GATSBY_FIREBASE_MESSAGING_SENDER_ID,
+          appId: process.env.GATSBY_FIREBASE_APP_ID,
+        });
```
Comment (lndgalante) [inline src/utils/firebase/client.js]: Can wee destructure env keys at the top?

```javascript
const { GATSBY_FIREBASE_API_KEY, GATSBY_FIREBASE_AUTH_DOMAIN, GATSBY_FIREBASE_DATABASE_URL, GATSBY_FIREBASE_PROJECT_ID, GATSBY_FIREBASE_STORAGE_BUCKET, GATSBY_FIREBASE_MESSAGING_SENDER_ID, GATSBY_FIREBASE_APP_ID } = process.env
```

```javascript
firebase.initializeApp({
  apiKey: GATSBY_FIREBASE_API_KEY,
  authDomain: GATSBY_FIREBASE_AUTH_DOMAIN,
  databaseURL: GATSBY_FIREBASE_DATABASE_URL,
  projectId: GATSBY_FIREBASE_PROJECT_ID,
  storageBucket: GATSBY_FIREBASE_STORAGE_BUCKET,
  messagingSenderId: GATSBY_FIREBASE_MESSAGING_SENDER_ID,
  appId: GATSBY_FIREBASE_APP_ID,
});
```

**Review by lndgalante — CHANGES_REQUESTED** [review]
Thanks @goncy great PR! I've left a few comments but change whatever you think necessary if you don't have too much time.

File: src/utils/firebase/client.js (hunk 5 above)
Comment (goncy) [reply src/utils/firebase/client.js:16]: So we can call a function everytime a method is called in an object without blocking its execution

**Review by lndgalante — APPROVED** [review]
Great work @goncy!! 🚀

**Pull request merged**